lib/improver/utilities/cube_manipulation.py

```diff
@@ -832,3 +833,36 @@ def enforce_coordinate_ordering(
     elif anchor == "end":
         cube.transpose(remaining_coords+coord_dims)
     return cube
+
+
+def clip_cube_data(cube, result):
+    """Apply np.clip to data in a resulting cube to ensure that the limits do
+    not go beyond the limits of the original input cube.
+
```

Comment:
This does more than just clip the data to the limits of the original input cube. It also sets the attributes and cell methods of the resulting cube to match those in the original cube.